src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.js

```diff
@@ -0,0 +1,237 @@
+import React, {
+    useContext, useEffect, useMemo,
+} from 'react';
+import {Dimensions, Keyboard} from 'react-native';
+import Reanimated, {
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming,
+} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import styles from '../../styles/styles';
+import {Actions, States, ActionSheetAwareScrollViewContext} from './ActionSheetAwareScrollViewContext';
+
+const config = {
```

Comment:
What's the story with this fancy bezier timing?

Reply:
This bezier timing represents something similar to how the keyboard works on iOS and Android. It's kinda in between these two animations. On iOS keyboard uses native spring which we can't really implement in React Native.

But this config is subject to change after I get more feedback from people testing this PR on their devices.

Reply:
Can't we use `useAnimatedKeyboard` to hook into the actual position of the native keyboard on a frame-by-frame basis?

Reply:
@roryabraham @terrysahaidak I think I've already mentioned it but the current implementation of `useAnimatedKeyboard` has a design flaw that introduces one frame of delay which is noticeable to some users and may cause UI flickers. The good news is that we have an idea how this should work and we are refactoring the native implementation of `useAnimatedKeyboard` which should hopefully eliminate the delay.

Reply:
I think it happens only sometimes randomly for me on iphone 11. 
I have more problems with modals when it has a constant 1-frame delay.
Basically, when you open a modal, the keyboard is closed without animation. And when you open the modal - it's back again. And now even though the state is correct the height has 1 frame delay, but i just use the previous height so it's not a big of an issue.
But that random flickering is.